src/main/java/com/ibm/g11n/pipeline/client/ServiceClient.java

```diff
@@ -441,6 +442,111 @@ public abstract void updateResourceEntry(String bundleId, String language,
+     *          [a-zA-Z0-9][a-zA-Z0-9_.-]* and the length must be less than or equal
+     *          to 255.
+     * @param type The type of document being created.
+     * @param newDocumentData
+     *          The new document's configuration.
+     * @throws ServiceException when the operation failed.
+     */
+    public abstract void createDocument(String documentId, DocumentType type,
+            NewDocumentData newDocumentData) throws ServiceException;
+
+    /**
+     * Returns the document's configuration.
+     * <p>
+     * This operation is only allowed to all user types, but only
+     * basic information (source language and target languages) is included
+     * in the result if the requesting user type is {@link UserType#READER READER}.
+     * of the service instance.
+     * 
+     * @param documentId  The document ID. Must not be null or empty.
+     * @param type  The type of document being requested.
+     * @return          The document's configuration.
+     * @throws ServiceException when the operation failed.
+     */
+    public abstract DocumentData getDocumentInfo(String documentId, DocumentType type) throws ServiceException;
+    
+    /**
+     * Updates the document's configuration.
+     * <p>
+     * This operation is only allowed to {@link UserType#ADMINISTRATOR ADMINISTRATOR}
+     * of the service instance.
+     * 
+     * @param documentId  The document ID.
+     * @param type The type of document to be updated.
+     * @param changeSet The change set of document configuration.
+     * @throws ServiceException when the operation failed.
+     */
+    public abstract void updateDocument(String documentId, DocumentType type,
+            DocumentDataChangeSet changeSet) throws ServiceException;
+
+    /**
+     * Update document content in the source language.
+     * <p>
+     * This operation is only allowed to {@link UserType#ADMINISTRATOR ADMINISTRATOR}
+     * of the service instance.
+     * 
+     * @param documentId  The bundle ID.
+     * @type type  The type of document being uploaded.
+     * @param language  The language specified by BCP 47 language tag.
+     * @param file A file containing the document to be uploaded.
+     * @throws ServiceException when the operation failed.
+     */
+    public abstract void updateDocumentContent(String documentId, DocumentType type,
+            String language, File file) throws ServiceException;
+
+    /**
+     * Retrieve the contents of a document in a particular language
+     * <p>
+     * This operation is only allowed to {@link UserType#ADMINISTRATOR ADMINISTRATOR}
+     * of the service instance.
+     * 
+     * @param documentId  The document ID.
+     * @type type  The type of document being requested.
+     * @param language  The language specified by BCP 47 language tag.
+     * @throws ServiceException when the operation failed.
+     */
+    public abstract byte[] getDocumentContent(String documentId, DocumentType type,
```

Comment:
Similar to above. It's OK to keep this, but I also want writeDocumentContents(... OutputStream os).